lib/ex_doc/autolink.ex

```diff
@@ -302,16 +303,32 @@ defmodule ExDoc.Autolink do
               {extra, "#" <> anchor}
           end
 
+        tool = tool_app(String.to_atom(app), config)
+
+        default_prefix =
+          case tool do
+            :otp ->
+              @otpappdocs
+
+            :ex_doc ->
+              @hexdocs
+          end
+
+        # TODO: In ex_doc_app_url we check if `app in config.apps` and
+        # then don't # link to hexdocs. Should we do the same here?
```

Comment:
Apps are only used by umbrella projects when generating docs for several projects at once. So in theory that could happen here if they have multiple Erlang sources. Could we use `ex_doc_app_url` here? Or would that be too complicated?

Reply:
Yeah, so maybe we don't use `tool_app` at all and instead write:

```
cond do
  app in config.apps -> # no prefix
  is_app_otp(app) -> @otpappdocs
  true -> @hexdocs
end
```

?

Reply:
Ah, so then maybe `e:` link prefix should not work for umbrella apps and they should use relative local links for extra pages?

Reply:
I think they should work because umbrella projects allow you generating docs both at the root (altogether) and individually, so linking should likely support both depending on config.apps?

Reply:
We also can write references to ourselves, right? So if `config.apps == [:stdlib]`, then `e:stdlib` should not have any prefix I think.

Reply:
Ah, ok. I've got no idea how umbrellas work so didn't know it would be that easy. I'll apply your suggest above after lunch and add a test or two for that case.

Reply:
So, just to clarify what we want:

```elixir
def prefix(app, config) do
  cond do
    app in config.apps ->
      ""
    Keyword.has_key?(app,config.deps) ->
     "#{Keyword.fetch!(app,config.deps)}"
    is_otp_app(app) ->
     @otpappdocs
    true ->
     @hexdocs
  end
end
```

How would I go about to warn for the case when an application that is not part of the dependencies is linked to? That is the code that currently does this:

```elixir
config.deps
|> Keyword.get_lazy(String.to_atom(app), fn ->
  maybe_warn(
    config,
    "documentation references \"e:#{string}\" but #{app} cannot be found in deps.",
    nil,
    %{}
    )
     @hexdocs <> "#{app}"
end)
```

I want to be able to catch misspellings for application names.

Reply:
Maybe this:

```elixir
def prefix(app, config) do
  cond do
    app in config.apps ->
      ""
    Keyword.has_key?(app,config.deps) ->
     "#{Keyword.fetch!(app,config.deps)}"
    is_otp_app(app) ->
     @otpappdocs
    is_elixir_app(app)
    true ->
     warn("omgomgomgomg")
     @hexdocs
  end
end
```

Reply:
I pushed an update that now works I think :) Basically the code became this:

```elixir
def prefix(app, config) do
  cond do
    app in config.apps ->
      Atom.to_string(app)
    Keyword.has_key?(app,config.deps) ->
     "#{Keyword.fetch!(app,config.deps)}"
    is_otp_app(app) ->
     @otpappdocs <> Atom.to_string(app)
    true ->
     if Application.load(app) != :ok do
        warn(....)
     end
     @hexdocs <> Atom.to_string(app)
  end
end
```

which is quite close to what you proposed.